### PR TITLE
ARTEMIS-1379 Some queue stats not threadsafe

### PR DESCRIPTION
**Pull request by jbertram**

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java [hunk 1]
```diff
@@ -177,9 +177,9 @@
 
    private final ScheduledDeliveryHandler scheduledDeliveryHandler;
 
-   private long messagesAdded;
+   private AtomicLong messagesAdded = new AtomicLong(0);
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java]: To be more memory friendly where lots of queues can the long be volatile and then use a AtomicLongFieldUpdater to safely update.

**Comment by michaelandrepearce** [comment]
Removed prev comment (as if needed should probably be a larger overhaul piece of work), and approved.

On another note, whilst not directly under the JIRA-1379, it looks like messagesExpired and messagesKilled have the same thread safe issue. Do you want to do under this ticket? Or want me to raise separate jira and me to do a separate PR?

**Comment by jbertram** [comment]
@michaelandrepearce, yes.  I'll update messagesExpired & messagesKilled as well.

**Comment by michaelandrepearce** [comment]
+1

**Pull request closed without merging**